### DISCUSSION TITLE
Setting default style for links?

### DISCUSSION THREAD

**Discussion by ronf** [Textualize/rich#977] [Q&A]
I'm looking for a way to set a default style (such as "blue underline") for links, without having to specify this style on every [link] I create. This seems like a job for a theme, and indeed you can create arbitrary named styles that can then be mapped to whatever you like, even redefining something like "blue" to a completely different color if you wanted to. However, it's not as easy to refer to "link" in a theme, since it requires a URL.

So far, I haven't found a way in a theme to write a generic entry which applies to all links, allowing the new theme to take a URL as an argument similar to how link does, but adding other styles defined in that theme. Here's the kind of thing I have in mind:

```python
    theme = Theme({'my_link': 'blue underline link'})
    console = Console(theme=theme)
    console.print('[my_link https://www.google.com]Google[/]')
```

**Comment by willmcgugan** [comment]
There isn't a simple way to do that currently. Maybe there should be. I'll give it some thought.

You could do it by post processing a Text instance, in a similar way to your link expander...

**Reply by ronf** [reply]
Hmm, good point. I already pass in the style for both the original link and the optional added URL when expanding links, so I could easily tack on an additional style there.
